apps/OboeTester/app/src/main/java/com/mobileer/oboetester/DeviceReportActivity.java

```diff
@@ -308,6 +316,118 @@ public String reportMediaCodecs() {
+            if (!file.exists()) {
+                return -1; // File not found
+            }
+            try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+                String line = br.readLine();
+                if (line != null) {
+                    return Integer.parseInt(line.trim());
+                }
+            } catch (IOException | NumberFormatException e) {
+                // Log.e(TAG, "Error reading " + path, e); // Use Android's Log if in an Android context
+                System.err.println("Error reading or parsing " + path + ": " + e.getMessage());
+            }
+            return -1; // Error or invalid data
+        }
+
+        /**
+         * Gathers detailed CPU core information and formats it into a string.
+         * This method requires appropriate file system permissions to access /sys paths.
+         *
+         * @return A neatly formatted string containing CPU core details, or an error message.
+         */
+        public static String reportCpuInfo() {
+            Map<Integer, CoreDetails> cpuDetailsMap = new TreeMap<>(); // Use TreeMap to keep cores sorted by index
+
+            int cpuIndex = 0;
+            boolean foundAnyCore = false;
+
+            while (true) {
+                String capacityPath = String.format(CPU_CAPACITY_PATH_FORMAT, cpuIndex);
+                String minFreqPath = String.format(CPU_MIN_FREQ_PATH_FORMAT, cpuIndex);
+                String maxFreqPath = String.format(CPU_MAX_FREQ_PATH_FORMAT, cpuIndex);
+                String curFreqPath = String.format(CPU_CURRENT_FREQ_PATH_FORMAT, cpuIndex);
+
+                // Check if at least one common file exists for this CPU index
+                if (!new File(capacityPath).exists() &&
+                        !new File(minFreqPath).exists() &&
+                        !new File(maxFreqPath).exists() &&
+                        !new File(curFreqPath).exists()) {
+                    // If none of the paths exist for this CPU index, assume no more CPUs
+                    break;
+                }
+                foundAnyCore = true;
+
+                // Get or create CoreDetails for the current CPU index
+                CoreDetails details = cpuDetailsMap.get(cpuIndex);
+                if (details == null) {
+                    details = new CoreDetails(cpuIndex);
+                    cpuDetailsMap.put(cpuIndex, details);
+                }
+
+                details.capacity = readSysfsInt(capacityPath);
+                details.minFreq = readSysfsInt(minFreqPath);
+                details.maxFreq = readSysfsInt(maxFreqPath);
+                details.currentFreq = readSysfsInt(curFreqPath);
+
+                cpuIndex++;
+            }
+
+            if (!foundAnyCore) {
+                return "############################\nCould not retrieve CPU core information. This usually means the necessary permissions are not available, or the /sys/devices/system/cpu paths do not exist on this device.";
```

Comment:
Very long line. Also, what are the necessary permissions? How can the user fix this?